spring-kafka/src/test/java/org/springframework/kafka/annotation/EnableKafkaIntegrationTests.java

```diff
@@ -247,7 +260,8 @@ public MultiListenerBean multiListener() {
 
 		@Bean
 		public ProducerFactory<Integer, String> producerFactory() {
-			return new DefaultKafkaProducerFactory<>(producerConfigs());
+			Map<String, Object> configs = producerConfigs();
+			return new DefaultKafkaProducerFactory<>(configs, producerStrategy(configs));
```

Comment:
May be you can take a look once again.  Even after this separation if you believe that this implementation is unnecessary, then I can think about simple Serializer/Deserializer injection directly into Factory classes.